replica/table.cc

```diff
@@ -1099,8 +1099,14 @@ compaction_group& table::compaction_group_for_key(partition_key_view key, const
 }
 
 compaction_group& tablet_storage_group_manager::compaction_group_for_sstable(const sstables::shared_sstable& sst) const noexcept {
-    auto [first_id, first_range_side] = storage_group_of(sst->get_first_decorated_key().token());
-    auto [last_id, last_range_side] = storage_group_of(sst->get_last_decorated_key().token());
+    size_t first_id, last_id;
+    locator::tablet_range_side first_range_side, last_range_side;
+    try {
+        std::tie(first_id, first_range_side) = storage_group_of(sst->get_first_decorated_key().token());
+        std::tie(last_id, last_range_side) = storage_group_of(sst->get_last_decorated_key().token());
+    } catch (std::out_of_range& e) {
+        on_internal_error(tlogger, format("Unable to load SSTable {} : {}", sst->get_filename(), e.what()));
+    }
```

Comment:
it looks like out_of_range exception is thrown only in storage_group_for_id() call below, so we may have to move decoration there.

Reply:
When I tried, the exception was thrown in this call only. If I move the try/catch below, the exception doesn't get caught.

Reply:
It's awkward because we only have the exception in storage_group_for_id()

storage_group& storage_group_manager::storage_group_for_id(const schema_ptr& s, size_t i) const {
    auto it = _storage_groups.find(i);
    if (it == _storage_groups.end()) [[unlikely]] {
        throw std::out_of_range(format("Storage wasn't found for tablet {} of table {}.{}", i, s->ks_name(), s->cf_name()));
    }
    return *it->second.get();
}

in release mode, storage_group_of(dht::token t) disables the checks. is it possible that your test triggered split and only one half of the sstable range was moved to another node with migration?

Reply:
I don't think so since you disabled balancing

Reply:
I tested in dev build, and I guess that's the reason I got the exception in the wrong place. I'll check and update the code.

Reply:
there's a fixture skip_mode(), you can perhaps use it to allow only release mode.

Reply:
since you don't really expect the decoration in the test, you can probably still run the test in all build modes.

Reply:
fixed it now.